docs/design.md

```diff
@@ -113,10 +113,18 @@ Drivers are essentially an extension of libnetwork and provides the actual imple
 * `driver.Join`
 * `driver.Leave` 
 
-These Driver facing APIs makes use of unique identifiers (`networkid`,`endpointid`,...) instead of names (as seen in user-facing APIs). 
+These Driver facing APIs makes use of unique identifiers (`networkid`,`endpointid`,...) instead of names (as seen in user-facing APIs).
 
 The APIs are still work in progress and there can be changes to these based on the driver requirements especially when it comes to Multi-host networking.
 
+### Driver semantics
+
+ * `Driver.CreateEndpoint`
+
+This method is passed an interface `EndpointInfo`, with methods `Interfaces` and `AddInterface`. If the slice returned by `Interfaces` is non-empty, the driver is expected to create the interfaces therein, and must return an error if it cannot. If the slice is empty, the driver should use `AddInterface` to provide zero or more fresh interfaces.
```

Comment:
@spikecurtis null and Host drivers are examples of that case.

@squaremo instead of `the driver is expected to create the interfaces`, I think the more appropriate statement would be to `the driver is expected to make use of the information`.